File: src/execution/runtimeWorkerEngine.ts
/**
 * Converted from Python: core/execution/runtime_worker_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function buildRuntimeWorkers(nodes: any): any {
  var workers: any[] = [];
  var index: any;
  var node: any;
  for ([index, node] of py.enumerate(py.slice(nodes, null, 1000))) {
    py.listAppend(workers, {"worker_id": py.toStr(py.get(node, "worker_id", py.get(node, "node_id", `worker:${py.toStr(index)}`))), "runtime": py.toStr(py.get(node, "runtime", "browser")), "synced": py.truthy(py.get(node, "synced", true)), "bounded": true});
  }
  return py.sorted(workers, {key: ((item: any) => py.at(item, "worker_id")) as (item: any) => any});
}

File: lib/src/execution/runtime_worker_engine.dart
/// Port of core/execution/runtime_worker_engine.build_runtime_workers.
List<Map<String, dynamic>> buildRuntimeWorkers(
    List<Map<String, dynamic>> nodes) {
  final List<Map<String, dynamic>> workers = <Map<String, dynamic>>[];
  final List<Map<String, dynamic>> bounded =
      nodes.length > 1000 ? nodes.sublist(0, 1000) : nodes;
  for (int index = 0; index < bounded.length; index++) {
    final Map<String, dynamic> node = bounded[index];
    workers.add(<String, dynamic>{
      'worker_id': '${node['worker_id'] ?? node['node_id'] ?? 'worker:$index'}',
      'runtime': '${node['runtime'] ?? 'browser'}',
      'synced': node.containsKey('synced') ? node['synced'] == true : true,
      'bounded': true,
    });
  }
  workers.sort((Map<String, dynamic> a, Map<String, dynamic> b) =>
      '${a['worker_id']}'.compareTo('${b['worker_id']}'));
  return workers;
}
